### PR TITLE
Enhancement / arena-core record index

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: core/record/_record/recordReader.js [hunk 1]
```diff
@@ -12,38 +12,28 @@ import * as Node from '../node'
...
 import { keys } from './recordKeys'
-import * as NodesIndex from './recordNodesIndex'
 
 /**
  * === simple getters
  */
 export const getNodes = R.propOr({}, keys.nodes)
 
-export const getNodeByUuid = (uuid) => R.path([keys.nodes, uuid])
+export const getNodeByUuid = (uuid) => (record) => Records.getNodeByUuid(uuid)(record)
```
Comment (ramirobg94) [inline core/record/_record/recordReader.js:21]:
```suggestion
export const getNodeByUuid = Records.getNodeByUuid
```

or even better


export { getNodeByUuid } form 'path to records ... '

File: core/record/_record/recordReader.js [hunk 2]
```diff
@@ -12,38 +12,28 @@ import * as Node from '../node'
...
 
-export const getRootNode = (record) => R.pipe(NodesIndex.getNodeRootUuid, (uuid) => getNodeByUuid(uuid)(record))(record)
+export const getRootNode = (record) => Records.getRoot(record)
```
Comment (ramirobg94) [inline core/record/_record/recordReader.js:23]:
```suggestion
export const getRootNode = Records.getRoot
```

File: core/record/_record/recordReader.js [hunk 3]
```diff
@@ -67,17 +57,10 @@ export const getAncestorByNodeDefUuid = (node, ancestorDefUuid) => (record) =>
   )(record)
 
 // Descendants
-export const getNodeChildren = (node) => (record) =>
-  R.pipe(
-    NodesIndex.getNodeUuidsByParent(Node.getUuid(node)),
-    R.map((uuid) => getNodeByUuid(uuid)(record))
-  )(record)
+export const getNodeChildren = (node) => (record) => Records.getChildren(node)(record)
```
Comment (ramirobg94) [inline core/record/_record/recordReader.js:60]:
```suggestion
export const getNodeChildren = Records.getChildren
```

File: core/record/_record/recordReader.js (hunk 1 above)
Comment (SteRiccio) [reply core/record/_record/recordReader.js:21]: I did it on purpose to be sure about the parameters ..

**Review by ramirobg94 — APPROVED** [review]

**Pull request merged**